### PR TITLE
Re-enable adaptive in embedded_torus BVP test (BVP MIRK 1.17)

### PR DESCRIPTION
**Pull request by ChrisRackauckas-Claude**
## Summary

Follow-up to #887. That PR pinned the `embedded_torus` test to `adaptive = false` because [SciML/BoundaryValueDiffEq.jl#484](https://github.com/SciML/BoundaryValueDiffEq.jl/issues/484) caused an `UndefRefError` in MIRK 1.15+ adaptive mesh refinement on nonlinear BVPs under the SciMLBase 3 / RAT 4 stack.

That upstream issue is now fixed in **`BoundaryValueDiffEqMIRK` 1.17.0**, registered earlier today (2026-05-01). I reproduced #484's standalone MWE on 1.17.0 — `MIRK4()` with default `adaptive = true` now returns `ReturnCode.Success` and lands on the correct boundary values.

This PR:

- Drops the `adaptive = false` workaround in `test/manifolds-old/embedded_torus.jl` so the test exercises the default (and more interesting) adaptive mesh-refinement path again.
- Bumps `BoundaryValueDiffEqMIRK` compat lower bound from `1.14` to `1.17` in `Project.toml`, guaranteeing the fix is present for anyone resolving this Manifolds version.
- Bumps Manifolds version `0.11.23` → `0.11.24` and adds a NEWS entry.

The pin and the linked comment are both gone; nothing else changed.

## Test plan

- [x] Reproduced #484's MWE on `BoundaryValueDiffEqMIRK 1.17.0` — `solve(bvp, MIRK4(); adaptive = true)` succeeds.
- [x] `test/manifolds-old/embedded_torus.jl` runs locally on Julia 1.11 with the workaround removed: `Torus in ℝ³ | 44 44 1m14.2s`.
- [x] Test env resolves `BoundaryValueDiffEqMIRK v1.17.0` / `BoundaryValueDiffEqCore v2.6.0` / `BoundaryValueDiffEq v5.23.0` cleanly.
- [ ] CI green.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name = "Manifolds"
 uuid = "1cead3c2-87b3-11e9-0ccd-23c62b72b94e"
 authors = ["Seth Axen <seth.axen@gmail.com>", "Mateusz Baran <mateuszbaran89@gmail.com>", "Ronny Bergmann <manopt@ronnybergmann.net>", "Antoine Levitt <antoine.levitt@gmail.com>"]
-version = "0.11.23"
+version = "0.11.24"
```
Comment (JoshuaLampert) [inline Project.toml]:
```suggestion
version = "0.11.23"
```
v0.11.23 is not yet released. The NEWS entry should be adapted accordingly.

**Review by mateuszbaran — APPROVED** [review]
